### PR TITLE
misc docs updates

### PR DESCRIPTION
**Pull request by joel-bluedata**
Heading toward the 0.5.0 release.

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+* Anyone building KubeDirector from source, as opposed to using a pre-built operator image, should re-read [doc/kubedirector-development.md](doc/kubedirector-development.md) to become acquainted with new requirements and with some tips for continuing to build your existing local repo. We highly recommend reading this **before** merging the new release source into your existing source.
+* Anyone who is currently running a KubeDirector 0.4.x deployment and is interested in updating that deployment in-place should read [doc/upgrade.md](doc/upgrade.md).
+
+KD 0.5.0 supports some additional properties in [KubeDirectorApp](https://github.com/bluek8s/kubedirector/wiki/KubeDirectorApp-Definition) and [KubeDirectorCluster](https://github.com/bluek8s/kubedirector/wiki/KubeDirectorCluster-Definition); see the "properties supported by KD v0.5.0+" sections in those tables. These implement three new features:
+* A KD app can indicate that a unique token should be generated to be used for authentication to a service endpoint. This token will be visible in the KD cluster member status, as well as visible within the relevant container (through configcli) for use in setting up that service.
```
Comment (kmathur2) [inline HISTORY.md:8]: the token is also available in service object's annotation

File: HISTORY.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
+# v0.5.0 - Jul ??, 2020
+
+The major change in this release is the move from operator SDK v0.8.1 to v0.15.2, which has a couple of notable sets of consequences:
+* Anyone building KubeDirector from source, as opposed to using a pre-built operator image, should re-read [doc/kubedirector-development.md](doc/kubedirector-development.md) to become acquainted with new requirements and with some tips for continuing to build your existing local repo. We highly recommend reading this **before** merging the new release source into your existing source.
```
Comment (koshiksamota10) [inline HISTORY.md:4]: Why do we have doc/kubedirector-development.md twice?

Comment (joel-bluedata) [reply HISTORY.md:4]: It's a markdown link. The text is "doc/kubedirector-development.md" and it links to the file doc/kubedirector-development.md.

File: doc/kd-release-process.md [hunk 3]
```diff
@@ -98,7 +103,7 @@ Delete the x.y.z-release branch everywhere (local, your GitHub, main GitHub).
 
 #### ADVERTISE THE RELEASE
 
-Modify the "current" wiki page documenting each CR as necessary to indicate that it is documenting a released API version. If development begins on a new API version in the future this text should be changed at that time.
+Modify the wiki page documenting each CRD so that it includes documentation of the support for any new properties added in this release. (Reference the existing tables to see how properties for a new version are added and marked.) Note that this is assuming the K8s API versioning practice where new properties are added to an existing API version while maintaining backwards compatibility; if we reach the point where a new API version is required then the CRD documentation will have to be further restructured.
```
Comment (kmathur2) [inline doc/kd-release-process.md:106]: Can any contributor create release?

**Comment by kmathur2** [comment]
Looks good overall 👍

File: doc/kd-release-process.md (hunk 3 above)
Comment (joel-bluedata) [reply doc/kd-release-process.md:106]: Nope, this is for those of us that have repo write and wiki write access. (And at the moment, really just for me.)

File: HISTORY.md (hunk 1 above)
Comment (joel-bluedata) [reply HISTORY.md:8]: Can you give me more detail about that? I can test later, but if you know off the top of your head that would be handy.

Comment (kmathur2) [reply HISTORY.md:8]: Let's say in the KDApp, jupyter notebook service of type NodePort has **hasAuthToken** set to True, then we generate this annotation in the svc Object
```
k8suser@kdss-tsxvh-0:~$ kubectl get svc s-kdss-4nbf8-0 -o yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    hpecp-internal-gateway/22: mip-bd-vm140.mip.storage.hpecorp.net:10006
    hpecp-internal-gateway/8888: mip-bd-vm140.mip.storage.hpecorp.net:10007
    kubedirector.hpe.com/kd-auth-token: 12df044ef8d1774c8c8772dafe6082b9
```

Comment (joel-bluedata) [reply HISTORY.md:8]: Added detail to the wiki; I'll also update this with a general comment.

**Pull request merged**